lib/ansible/modules/packaging/language/pip.py

```diff
@@ -302,6 +302,15 @@ def _recover_package_name(names):
     return package_names
 
 
+def _normalize_package_name(name):
```

Comment:
This function does not behave the some way pip does. It would be better to use the some underlying code, see 
https://packaging.pypa.io/en/latest/_modules/packaging/utils/#canonicalize_name

It would be best to import this from some core library, but as far as I see it's not exposed, so for me it would also be ok to copy the code here with a reference to the source.

Reply:
Nice find will add it shortly. Thx!